### PR TITLE
[NIFI-9554] update node/npm versions and frontend deps

### PR DESCRIPTION
**Pull request by scottyaslan**
To test:

Please install and use node v16.13.2. In the root of the projext run `npm run ci && npm run watch` and verify the available UX components are available and that the styles match nifi-fds v0.2.0.

NOTE: There will be some warnings displayed in the terminal and in the browser and in the browser console. Those are webpack dev server telling us that some of our deps are not providing licenses. I couldn't find a good way to silence those but they can be ignored as they will only show up in dev mode. Simply close the warning overlay and you should be able to verify the UX/UI components.

Next, run `npm run clean:install` and verify tests pass.

Thank you for submitting a contribution to Apache NiFi Flow Design System.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with either NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you ensured that a full build and that the full suite of unit tests is executed via npm run clean:install at the root nifi-fds folder?
- [ ] Have you written or updated the Apache NiFi Flow Design System demo application to demonstrate any new functionality, provide examples of usage, and to verify your changes via npm start at the nifi-fds/target folder?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-fds?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-fds?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: platform/core/snackbars/fds-snackbars.module.js [hunk 1]
```diff
@@ -38,6 +32,10 @@ import {
 } from './services/snackbar.service';
 
 import { FdsCoasterComponent } from './coaster/coaster.component';
+import {MatSnackBarModule} from "@angular/material/snack-bar";
```
Comment (sardell) [inline platform/core/snackbars/fds-snackbars.module.js:35]: This is nit picky, but I think the imports statements should be formatted like the other existing statements with a space between the curly brackets and the module name.

Comment (scottyaslan) [reply platform/core/snackbars/fds-snackbars.module.js:35]: ty... not sure why the linter didn't complain...

**Review by mtien-apache — APPROVED** [review]
Works well! 👍🏼

**Review by sardell — APPROVED** [review]
Aside from the import styling I pointed out, this PR looks good to me. +1 (non-binding) pending that being fixed.

**Review by rfellows — APPROVED** [review]
LGTM 👍 . Thanks @scottyaslan

**Review by mcgilman — APPROVED** [review]
Thanks @scottyaslan! +1

File: platform/core/snackbars/fds-snackbars.module.js (hunk 1 above)
Comment (scottyaslan) [reply platform/core/snackbars/fds-snackbars.module.js]: ok I updated the linter and addressed the lint errors it found. plz confirm

File: .eslintrc.js [hunk 2]
```diff
@@ -15,77 +15,110 @@
...
+        'react/no-this-in-sfc': OFF,
+        'prefer-promise-reject-errors': OFF,
+        'object-curly-newline': OFF,
+        'no-restricted-globals': OFF,
+        'import/prefer-default-export': OFF,
+        'linebreak-style': OFF,
+        'quotes': [ERROR, 'single', { 'allowTemplateLiterals': true }],
+        'no-useless-constructor': OFF,
+        'no-empty-function': OFF,
+        'no-underscore-dangle': ['error', { 'allowAfterThis': true }]
```
Comment (rfellows) [inline .eslintrc.js:122]: shouldn't this be:
```suggestion
        'no-underscore-dangle': [ERROR, { 'allowAfterThis': true }]
```

File: platform/core/common/fds.animations.js [hunk 3]
```diff
@@ -31,88 +31,185 @@ FdsAnimations.prototype = {
...
+                        style(
+                            {
+                                opacity: 0
+                            }
+                        )
+                    )
+                ]
+            )
+        ]
+    ),
```
Comment (rfellows) [inline platform/core/common/fds.animations.js:70]: Was this changed to satisfy the formatting rules? Looked better before IMO :(

**Review by rfellows — APPROVED** [review]
Changes look good! +1

**Pull request merged**